### DISCUSSION TITLE
GitHub is not opening in any browser

### DISCUSSION THREAD

**Discussion by sumitsarkardotAI** [community/community#117214] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

My GitHub is now opening in any browser however I performed all the necessary steps to fix the issue but couldn't. When I used VPN to do the same GitHub opened. Please help me fix this issue.
![image_2024-04-06_200722753](![Screenshot (18)](https://github.com/community/community/assets/61076636/e2f7e070-5865-4241-80a0-86e97e55a0ed)

**Comment by anouar-garbaa** [comment]
hello,
Your network might be interfering with accessing GitHub so i think trying those steps may solve the problem:
* Try changing your DNS settings to a public DNS provider like Google DNS (8.8.8.8 & 8.8.4.4) and see if GitHub opens.
* Try temporarily disabling all browser extensions and see if it resolves the issue
* Try clearing your browser's cache and cookies and see if it helps.
* Check your firewall or proxy settings for any rules blocking GitHub.
i hope it works with you

**Reply by sumitsarkardotAI** [reply]
Actually these steps I have tried but still don't work, can you please provide me a live support with it.

**Reply by anouar-garbaa** [reply]
I had a similar problem five months ago, but it was solved by simply disabling all browser extensions !!
It's strange that the problem hasn't been resolved yet :(

**Comment by sumitsarkardotAI** [comment]
Is there any live support feature for this thing as I need it to work
urgent.

**Comment by eliz-amst** [comment]
Open console in your browser, navigate to network tab and check what status code you are getting so you can see what is the root of the problem and where is it coming from, after you check the status code we can go from there and see steps to troubleshoot

**Reply by sumitsarkardotAI** [reply]
Getting this error:  (failed)net::ERR_CONNECTION_REFUSED
![Uploading Screenshot (21).png…]()

**Reply by sumitsarkardotAI** [reply]
![Screenshot (21)](https://github.com/community/community/assets/61076636/0d4c6745-ebf6-490f-9673-66578a754c25)

**Reply by eliz-amst** [reply]
Clear your browser's cache, if you're using a VPN or proxy, try disabling it temporarily to see if it resolves the issue, make sure that GitHub's domain is not blocked or redirected in your computer's hosts file, flush your computer's DNS cache to ensure that it's resolving domain names correctly. You can do this by opening the command prompt or terminal and running the appropriate command for your operating system. For example, on Windows, you can run ipconfig /flushdns. Make sure that your firewall or antivirus software is not blocking connections to GitHub. If you're still experiencing issues, try restarting your router to reset your network connection.

**Reply by sumitsarkardotAI** [reply]
Thanks will try an will update if there is a change

**Reply by eliz-amst** [reply]
If none of these steps resolve the issue, it's possible that there may be a problem with your ISP or GitHub's servers. In that case, you may need to contact your ISP for further assistance, or wait for GitHub to resolve any server-side issues. Have you contacted them?

**Comment by sumitsarkardotAI** [comment]
Getting this error:  (failed)
net::ERR_CONNECTION_REFUSED


Sumit Sarkar

**Reply by NikhilDinesan11** [reply]
Im also getting the same error as you are...was there any fix for this ??

**Comment by RFAhmadi** [comment]
My suggestion is to clear your browser history. Caches and cookies. Media Foundation data, Site permissions. But there is no need to erase the rest. Be careful NOT to delete saved passwords.

**Reply by sumitsarkardotAI** [reply]
what is Media Foundation data? and how to clear it

**Reply by eliz-amst** [reply]
Media Foundation is a framework developed by Microsoft for multimedia processing on Windows operating systems. It provides functions and APIs for encoding, decoding, and processing audio and video streams. Media Foundation data refers to the data and metadata associated with multimedia files and streams that are processed using Media Foundation.
Clearing Media Foundation data typically involves clearing the caches and settings related to multimedia processing on your system.

**Reply by RFAhmadi** [reply]
Another reason I think is involved, check what your proxy is set to.
and turn it off.
stteing > Network and internet > proxy
![Screenshot 2024-04-24 203009](https://github.com/community/community/assets/107707063/c83bfd95-1ff2-45d6-a44c-0c2075198a91)

**Reply by RFAhmadi** [reply]
It's all there.
Where you clear the browser history, if you scroll a little you will see it.
![Screenshot 2024-04-24 203529](https://github.com/community/community/assets/107707063/ec2c1666-e61c-488a-9dbe-c8531cc17136)

**Reply by RFAhmadi** [reply]
wait! 
If GitHub doesn't open, then how do you respond and see messages? 😂😅

**Reply by sumitsarkardotAI** [reply]
I am using VPN bro😂, while turning vpn on I can use it. Also in my mobile I'm using the same internet connection but can access it there. The issue is in the laptop itself which I'm trying to fix

**Reply by RFAhmadi** [reply]
I see now. So clearing the browser history itself should most likely solve the problem. It seems to be a problem with Microsoft Edge, because I had the same problem with Twitter and Instagram recently.

**Comment by sumitsarkardotAI** [comment]
No, I have not. I couldn't find the steps to do so as it was transferring
me to the community page

Sumit Sarkar

**Reply by eliz-amst** [reply]
under this community tabs there is a button to open the ticket if the community pages are not helpful

**Reply by sumitsarkardotAI** [reply]
thanks for the information I'm doing it right away

**Reply by sumitsarkardotAI** [reply]
Actually for me the open ticked option is not displayed. Can you guide me how to open one.

**Reply by eliz-amst** [reply]
https://support.github.com/request

choose a topic and fill out the request, you will be able to see your tickets after they are submitted and get notified when answered

**Comment by sumitsarkardotAI** [comment]
yes it is off

Sumit Sarkar

**Comment by sumitsarkardotAI** [comment]
No bro didn't get any perfect solution till now

**Reply by NikhilDinesan11** [reply]
Thanks for the update...are you still using VPN ? and did you contact your ISP ??

**Comment by sumitsarkardotAI** [comment]
yes bro using vpn😂 till now

**Reply by NikhilDinesan11** [reply]
Hey i talked to my ISP and they did some changes from their end...its working!!!

**Reply by nerdy766** [reply]
U contacted airtel customer care

**Comment by sumitsarkardotAI** [comment]
who is your isp

**Reply by NikhilDinesan11** [reply]
Airtel

**Comment by MarkCyber** [comment]
Could be a firewall issue preventing access to the website, so I would check fw configuration or maybe your hosts file at c:\Windows\System32\Drivers\etc\hosts on windows or etc/hosts on linux. However, considering you now have access after using your VPN, it is possible that for some reason you are IP banned. I would try changing your IP address and seeing if you can connect after getting a new IP.

**Comment by sumitsarkardotAI** [comment]
can you help me what do to and how to fix that issue,as I guess you have
faced this issue before and got the perfect solution for the same.

**Comment by MarkCyber** [comment]
> can you help me what do to and how to fix that issue,as I guess you have faced this issue before and got the perfect solution for the same.

Sure!

Regarding your IP address, there are a few ways to change it. You have a public and a private IP address. To change your public IP address, it is as simple as restarting your router. This can be done by unplugging it from the power source, waiting 5-10 minutes and plugging it back in. You may also call your Internet Service Provider to change your public IP address if necessary.

To change your private IP address on a windows device, navigate to the network and sharing section. From there click on "change adapter settings" and then, right click on your current connection and go to "properties". From there, go to internet protocol version 4 (TCP/IPv4) and once again, click on "properties". From there you should be able to change your IP address to your desired address.

I hope this helps. Let me know!

**Comment by sumitsarkardotAI** [comment]
brother in that list github is written how do I remove it

**Reply by MarkCyber** [reply]
> brother in that list github is written how do I remove it
In which list are you referring to?

**Comment by sumitsarkardotAI** [comment]
in hosts file github is written

**Reply by NikhilDinesan11** [reply]
i guess you can comment it out using #..check chat gpt also

**Reply by MarkCyber** [reply, marked as answer]
Add a # before it to comment it out, like this:
'#' 127.0.0.1 GitHub.com
without the ' quotes before and after the #

**Reply by sumitsarkardotAI** [reply]
> Add a # before it to comment it out, like this: '#' 127.0.0.1 GitHub.com without the ' quotes before and after the #

Mark, I'm quoting your answer in this comment: 

Could be a firewall issue preventing access to the website, so I would check fw configuration or maybe your hosts file at c:\Windows\System32\Drivers\etc\hosts on windows or etc/hosts on linux. However, considering you now have access after using your VPN, it is possible that for some reason you are IP banned. I would try changing your IP address and seeing if you can connect after getting a new IP.

**Comment by sumitsarkardotAI** [comment]
yes but I found an answer to it, I firstly opened Notepad as admin and then
opened the file using file>open, and then made the changes and saved it.
Just now I'm going to check if it is working or not.

**Reply by MarkCyber** [reply]
Good luck! Let me know :)

**Reply by Zulipan7** [reply]
what change do you make? I follow your direction to system32 and host and then what did you do?

**Comment by sumitsarkardotAI** [comment]
yes mark

**Comment by sumitsarkardotAI** [comment]
sure

**Comment by sumitsarkardotAI** [comment]
sure, I'll check once I confirm the previous process is working or not

**Comment by sumitsarkardotAI** [comment]
Guys, the issue is fixed. I'm writing this message from my browser without using VPN.

**Reply by MarkCyber** [reply]
I'm happy to hear that!!! :)

**Comment by sumitsarkardotAI** [comment]
Thanks for the support
Sumit Sarkar

**Comment by DebdutBiswas** [comment]
**I don't know why GitHub hasn't moved to IPv6 yet, it's 2024**.

I have noticed and debugged this same issue. It is because of github's load balancer might be doing some TCP packet filtering. It is happening mostly from the **464XLAT** based networks. Which now a days 5G mobile networks are moving from dual stack IPv6 + IPv4 to IPv6 only networks. Customer CPE / Router or end device does the **464XLAT** translation to convert the IPv4 literals from local network or from the end device itself to routable packet via **NAT64** translation server. That is set up by ISPs or service providers. I have tested GitHub access behind from various 464XLAT networks and noticed strange issue ipv4 addresses from **20.207.x.x** (Bangalore, Singapore, London) and **4.237.x.x** (Sydney) ranges are pingable but not able to complete the TLS handshakes and getting TCP connection timeouts but ip ranges from **140.82.x.x** (Amsterdam, Frankfurt, New York, San Francisco) of GitHub's load balancer is able to open TCP connections and TLS handshake completes and getting https responses fine. Noticed similar behavior from Reddit.com. Every other IPv4 only services are working fine, Discord is working fine with WebRTC calls and chats which requires end to end IPv4.

**Comment by DebdutBiswas** [comment]
I finally fixed it, GitHub load balancer in some regions rejects fragmented packets, so I had to adjust **MSS** on the outgoing network interface. Now GitHub is working fine from behind **464XLAT** networks.

**Reply by yawnyoung** [reply]
which mss do you adjust to? thanks!

**Comment by sumitsarkardotAI** [comment]
thanks so much

**Comment by edwarddusingeze-cyber** [comment]
> open
